internal/service/vmservice/ip.go

```diff
@@ -74,13 +74,13 @@ func formatIPAddressName(name, device string) string {
 	return fmt.Sprintf("%s-%s", name, device)
 }
 
-func machineHasIPAddress(machine *infrav1alpha1.ProxmoxMachine) bool {
-	return machine.Status.IPAddresses[infrav1alpha1.DefaultNetworkDevice] != (infrav1alpha1.IPAddress{})
+func machineHasIPAddress(machine *infrav1.ProxmoxMachine) bool {
+	return machine.Status.IPAddresses[infrav1.DefaultNetworkDevice] != (infrav1.IPAddress{})
 }
 
 func handleIPAddressForDevice(ctx context.Context, machineScope *scope.MachineScope, device, format string, ipamRef *corev1.TypedLocalObjectReference) (string, error) {
```

Comment:
Can this function be refactored so that it always returns an error when it also returns an empty string? To avoid all the `if err != nil || ip == "" {`.

Reply:
It's very confusing to return an error when creating a new IP claim:

https://github.com/ionos-cloud/cluster-api-provider-proxmox/blob/0cd195bb54ec8ddd57c80061676d77e82fc1d6d0/internal/service/vmservice/ip.go#L94-L98

 also when trying to add a tag:

https://github.com/ionos-cloud/cluster-api-provider-proxmox/blob/0cd195bb54ec8ddd57c80061676d77e82fc1d6d0/internal/service/vmservice/ip.go#L112-L117

Another choice is to add a third return as bool, but it's actually the same.